### PR TITLE
Adding amount comparison and amount identification functions

### PR DESCRIPTION
**Pull request by dotpointer**
Adding amount comparison and amount identification functions, as requested in https://github.com/Neovici/cosmoz-frontend/pull/544.

### CODE REVIEW

File: cosmoz-moneyhelper-behavior.html [hunk 1]
```diff
@@ -112,6 +112,30 @@
...
 	 */
 	Cosmoz.MoneyHelperBehavior = {
+		/**
+		 * Compare amounts.
+		 * @param {object} amount1 Amount 1.
+		 * @param {object} amount2 Amount 2.
+		 * @param {number} precision Decimal precision, defaults to 2 decimals.
+		 * @returns {boolean} Comparison result.
+		 */
+		amountEquals(amount1, amount2, precision = 2) {
```
Comment (nomego) [inline cosmoz-moneyhelper-behavior.html]: Could also be as beautiful as below (arrow)

File: cosmoz-moneyhelper-behavior.html [hunk 2]
```diff
@@ -112,6 +112,30 @@
...
+			return this.round(amount1.amount, precision) === this.round(amount2.amount, precision) &&
```
Comment (nomego) [inline cosmoz-moneyhelper-behavior.html]: Needs null-checks for amount1 and amount2 - use `this.isAmount()`!

File: cosmoz-moneyhelper-behavior.html [hunk 3]
```diff
@@ -112,6 +112,30 @@
...
+				amount1.currency === amount2.currency;
+		},
+		/**
+		 * Determine if a constant or a variable is an amount.
+		 * @param {object} potentialAmount Potential amount.
+		 * @returns {boolean} Result of determination.
+		 */
+		isAmount: potentialAmount =>
+			potentialAmount !== null &&
```
Comment (nomego) [inline cosmoz-moneyhelper-behavior.html]: Should include `undefined` - `!= null`

File: cosmoz-moneyhelper-behavior.html [hunk 4]
```diff
@@ -112,6 +112,30 @@
...
+					typeof potentialAmount === 'object' &&
```
Comment (nomego) [inline cosmoz-moneyhelper-behavior.html]: Check not needed, if `potentialAmount != null` we can try read `amount`

File: cosmoz-moneyhelper-behavior.html [hunk 5]
```diff
@@ -112,6 +112,30 @@
...
+					potentialAmount.amount != null &&
+					typeof potentialAmount.amount === 'number' &&
+					potentialAmount.currency &&
```
Comment (nomego) [inline cosmoz-moneyhelper-behavior.html]: `!= null`

File: cosmoz-moneyhelper-behavior.html (hunk 3 above)
Comment (nomego) [inline cosmoz-moneyhelper-behavior.html]: Could all checks have this indent or will lint complain?

File: cosmoz-moneyhelper-behavior.html [hunk 6]
```diff
@@ -112,6 +112,30 @@
...
+		amountEquals(amount1, amount2, precision = 2) {
+			return this.round(amount1.amount, precision) === this.round(amount2.amount, precision) &&
+				amount1.currency === amount2.currency;
+		},
+		/**
+		 * Determine if a constant or a variable is an amount.
+		 * @param {object} potentialAmount Potential amount.
+		 * @returns {boolean} Result of determination.
+		 */
+		isAmount: potentialAmount =>
```
Comment (nomego) [inline cosmoz-moneyhelper-behavior.html]: Let's do similar checks in order for readability, null-checks, type-checks, length

File: cosmoz-moneyhelper-behavior.html [hunk 7]
```diff
@@ -112,6 +112,30 @@
 	 * @demo
 	 */
 	Cosmoz.MoneyHelperBehavior = {
+		/**
+		 * Compare amounts.
+		 * @param {object} amount1 Amount 1.
+		 * @param {object} amount2 Amount 2.
+		 * @param {number} precision Decimal precision, defaults to 2 decimals.
+		 * @returns {boolean} Comparison result.
```
Comment (nomego) [inline cosmoz-moneyhelper-behavior.html]: Be more expressive.. what's the comparison result? If they equal eachother? Then write that rather.

File: cosmoz-moneyhelper-behavior.html [hunk 8]
```diff
@@ -112,6 +112,30 @@
...
+		 */
+		amountEquals(amount1, amount2, precision = 2) {
+			return this.round(amount1.amount, precision) === this.round(amount2.amount, precision) &&
+				amount1.currency === amount2.currency;
+		},
+		/**
+		 * Determine if a constant or a variable is an amount.
+		 * @param {object} potentialAmount Potential amount.
+		 * @returns {boolean} Result of determination.
```
Comment (nomego) [inline cosmoz-moneyhelper-behavior.html]: be more expressive! :)

File: cosmoz-moneyhelper-behavior.html (hunk 2 above)
Comment (dotpointer) [reply cosmoz-moneyhelper-behavior.html]: Updated with use of that function.

File: cosmoz-moneyhelper-behavior.html (hunk 3 above)
Comment (dotpointer) [reply cosmoz-moneyhelper-behavior.html]: Removed one =.

File: cosmoz-moneyhelper-behavior.html (hunk 4 above)
Comment (dotpointer) [reply cosmoz-moneyhelper-behavior.html]: Removed check.

File: cosmoz-moneyhelper-behavior.html (hunk 5 above)
Comment (dotpointer) [reply cosmoz-moneyhelper-behavior.html]: Added `!= null` to currency check.

File: cosmoz-moneyhelper-behavior.html (hunk 3 above)
Comment (dotpointer) [reply cosmoz-moneyhelper-behavior.html]: Seems to pass with 2 tabs.

File: cosmoz-moneyhelper-behavior.html (hunk 6 above)
Comment (dotpointer) [reply cosmoz-moneyhelper-behavior.html]: Sorted them in null, type and length checks.

File: cosmoz-moneyhelper-behavior.html (hunk 7 above)
Comment (dotpointer) [reply cosmoz-moneyhelper-behavior.html]: Added a more descriptive text.

File: cosmoz-moneyhelper-behavior.html (hunk 8 above)
Comment (dotpointer) [reply cosmoz-moneyhelper-behavior.html]: Added a more descriptive text here too.

**Pull request merged**